akka-remote/src/main/scala/akka/remote/artery/ArteryTransport.scala

```diff
@@ -116,7 +117,39 @@ private[akka] trait InboundContext {
  */
 private[akka] object AssociationState {
   def apply(): AssociationState =
-    new AssociationState(incarnation = 1, uniqueRemoteAddressPromise = Promise(), quarantined = Set.empty)
+    new AssociationState(incarnation = 1, uniqueRemoteAddressPromise = Promise(), quarantined = QuarantinedUidSet.empty)
+
+  object QuarantinedUidSet {
```

Comment:
I did this because of boxing from ordinary `Set`.
I will write an array based immutable `Map` for long keys anyway for the UID -> Association and then I will probably replace the `QuarantinedUidSet` with that one.